cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/EndPointStage.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.api;
+
+import co.cask.cdap.templates.etl.api.config.ETLStage;
+
+/**
+ * Base class for ETL Source, Sink Stages.
+ */
+public class EndPointStage {
```

Comment:
since configure is common for all stages, I think it makes more sense for this to be an abstract class called Stage and have source, sink, transforms extend it.  We can move configurePipeline to source and sink.  Then the code to configure sources, sinks, and transforms is all the same rather than one way for sources and sinks, and another for transforms.